pkg/cluster/cluster.go

```diff
@@ -168,6 +173,66 @@ func (c *cluster) ensureSSHKey() error {
 	)
 }
 
+// ensureHostKey records the host key when needed
+func (c *cluster) ensureHostKey(machine *Machine) error {
+	if c.config.Cluster.KnownHosts == "" {
+		return nil
+	}
+	path, _ := homedir.Expand(c.config.Cluster.KnownHosts)
+
+	utils.Logger.Infof("Getting SSH key: %s ...", machine.machineName)
+	hostline, err := machine.HostKey()
+	if err != nil {
+		return err
+	}
+	hostline = strings.TrimSuffix(hostline, "\n")
+	fields := strings.Fields(hostline)
+	hostkey := fields[2]
+	if _, err := os.Stat(path); err != nil {
+		return os.WriteFile(path, []byte(hostline), 0o644)
+	}
+
+	// check if the host key is already in known_hosts
+	r, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+	known := false
+	lines := []string{}
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == hostline {
+			known = true
+			continue
+		}
+		fields := strings.Fields(line)
+		if fields[2] == hostkey {
+			//remove
```

Comment:
This code doesn't work if multiple hosts are using the same key, for that it needs to check for known hosts.

i.e. check `fields[0]` against a list of all machines, and see if someone else in the cluster is using this line...